services/registration/src/routes/application.ts

```diff
@@ -252,6 +255,26 @@ applicationRouter.route("/:id/actions/submit-application").post(
       })
     );
 
+    if (
+      branchType === "APPLICATION" &&
+      autoConfirm?.enabled &&
+      ((autoConfirm.emails ?? []).includes("*") || // matches all emails
+        (autoConfirm.emails ?? []).includes(existingApplication.email) || // matches complete emails
+        (autoConfirm.emails ?? []).includes(existingApplication.email.replace(/^[^@]*@/, ""))) // matches emails by domain
```

Comment:
I'm not sure if this regex will match the email domain. Try inputting it into this tester (https://regex101.com/). Ideally, if someone has a email like john@ncr.com, the domain we'd want to match is @ncr.com